examples/annotation/detect_hijacks.py

```diff
@@ -25,58 +32,105 @@ def choose_input(input):
         raise ValueError("unknown input type {}".format(input))
 
 
-if __name__ == "__main__":
-    import argparse
+def registry_kwargs(file_path):
+    if os.path.basename(file_path) == "routes.csv":
+        kwargs["irr_ro_file"] = file_path
+
+    if os.path.basename(file_path) == "roa.csv":
+        kwargs["rpki_roa_file"] = file_path
+
+    if os.path.basename(file_path) == "maintainers.csv":
+        kwargs["irr_mnt_file"] = file_path
+
+    if os.path.basename(file_path) == "organisations.csv":
+        kwargs["irr_org_file"] = file_path
+    
+    return kwargs
+
+
+class PollingHandler(FileSystemEventHandler):
+    
+    def on_created(self, event):
+        super(PollingHandler, self).on_created(event)
+        what = 'directory' if event.is_directory else 'file'
+        logging.info("Created %s: %s", what, event.src_path)
+
+        if args.registry_path == os.path.dirname(event.src_path):
+            """If an event is triggering the registry path 
+            then load the registry data in structures and radix trees"""
+
+            reg_kwargs = registry_kwargs(event.src_path)
+
+            if len(reg_kwargs) == 4:
+                self.list_funcs = parse_registry_data(**reg_kwargs)
+                logging.info("Completed parsing registry data")
+                       
+        if args.bgp_path == os.path.dirname(event.src_path):
+            """If an event is triggering bgp path, read files
+             and load them into radix trees """
+
+            mrt_files.append(event.src_path)
+            input_kwargs = {"files": mrt_files}
+            input = choose_input(args.input)
+            bgp_kwargs = input(args.collector, **input_kwargs)
+
+            for conflict in detect_hijacks(self.list_funcs, **bgp_kwargs):
```

Comment:
One solution might be to look for the file and if present, process it here. This will handle the restart case.